ext/opcache/Optimizer/dfa_pass.c

```diff
@@ -992,6 +992,85 @@ void zend_dfa_optimize_op_array(zend_op_array *op_array, zend_optimizer_ctx *ctx
 		ssa_verify_integrity(op_array, ssa, "before dfa");
 #endif
 
+		/* Split series of assignments */
+		for (v = op_array->last_var; v < ssa->vars_count; v++) {
+			int op_2, v_2;
+			zend_op *opline2;
+
+			if (ssa->vars[v].var < op_array->last_var
+			 || ssa->vars[v].definition < 0
+			 || ssa->ops[ssa->vars[v].definition].result_def != v
+			 || ssa->vars[v].use_chain < 0
+			 || ssa->vars[v].phi_use_chain != NULL
+			 || ssa->vars[v].sym_use_chain != NULL) {
+				continue;
+			}
+
+			op_1 = ssa->vars[v].definition;
+			opline = op_array->opcodes + op_1;
+
+			if (opline->result_type != IS_VAR
+			 || (opline->opcode != ZEND_ASSIGN
+			  && opline->opcode != ZEND_ASSIGN_OP
+			  && opline->opcode != ZEND_PRE_INC
+			  && opline->opcode != ZEND_POST_INC)
+			 || opline->op1_type != IS_CV) {
+				continue;
+			}
+
+			op_2 = ssa->vars[v].use_chain;
+			opline2 = op_array->opcodes + op_2;
+			v_2 = ssa->ops[op_1].op1_def;
+
+			if (ssa->vars[v_2].use_chain >= 0
+			 && ssa->vars[v_2].use_chain <= op_2) {
+			 continue;
+			}
+
+			if (ssa->ops[op_2].op1_use == v
+			 && ssa->ops[op_2].op1_use_chain < 0
+			 && opline2->opcode != ZEND_SEND_REF
+			 && opline2->opcode != ZEND_SEND_VAR
+			 && opline2->opcode != ZEND_SEND_VAR_EX
+			 && opline2->opcode != ZEND_SEND_VAR_NO_REF
+			 && opline2->opcode != ZEND_SEND_VAR_NO_REF_EX
+			 && opline2->opcode != ZEND_RETURN_BY_REF) {
```

Comment:
This blacklist looks fragile. For example, do we also need to handle by-ref ZEND_YIELD? Can this pattern occur together with ZEND_CASE (`switch ($a = $b)`)?

Reply:
ZEND_CASE is covered by check of def/use chain for single use.